### PR TITLE
Remove global variables #50

### PR DESCRIPTION
**Pull request by andrea841**
- created 2 fields in DomainPool replacing global variables 
- changed code calculating number of spaces to leave for the print statement on number of generated sequences (it used to be based on the constant with value 10 ** 5, but now changes based on what the variable is)

### CODE REVIEW

File: dsd/constraints.py [hunk 1]
```diff
@@ -611,6 +610,19 @@ class DomainPool:
...
     """
 
+    generation_upper_limit: int = 4 ** 12
+    """
+    Maximum number of sequences to generate at once. If :py:data:`DomainPool.current_generation_amount` 
+    exceeds this number, the user is asked to re-adjust constraint strictness. The number may be changed to 
+    satisfy individual device capabilities.  
+    """
+
+    current_generation_amount: int = field(default=10**5, init=False)
```
Comment (dave-doty) [inline dsd/constraints.py:620]: I'd rename this field to `num_sequences_to_generate`, and remove `init=False`.

**Review by dave-doty — CHANGES_REQUESTED** [review]

File: dsd/constraints.py [hunk 2]
```diff
@@ -68,7 +68,7 @@
 replace_with_close_sequences_key = 'replace_with_close_sequences'
 hamming_probability_key = 'hamming_probability'
 generation_upper_limit_key = 'generation_upper_limit'
```
Comment (dave-doty) [inline dsd/constraints.py:70]: Sorry, for consistency, can we rename `generation_upper_limit` to `num_sequences_to_generate_upper_limit`?

File: dsd/constraints.py [hunk 3]
```diff
@@ -88,7 +88,7 @@ def default_score_transfer_function(x: float) -> float:
     return max(0.0, x ** 2)
 
 # Commented code moved to DomainPool._generate_sequences_satisfying_numpy_constraints due to use of the variable
```
Comment (dave-doty) [inline dsd/constraints.py:90]: You can just remove this commented out code.

**Review by dave-doty — CHANGES_REQUESTED** [review]

File: dsd/constraints.py (hunk 2 above)
Comment (andrea841) [reply dsd/constraints.py:70]: Sure!

**Review by dave-doty — APPROVED** [review]

**Pull request merged**